src/ServiceControl/Recoverability/Retries/Infrastructure/ReturnToSenderDequeuer.cs

```diff
@@ -1,25 +1,120 @@
+            }
+            catch (Exception)
+            {
+                message.Headers["ServiceControl.TargetEndpointAddress"] = destination;
+
+                throw;
+            }
+        }
+
+        protected override IManageMessageFailures FaultManager
+        {
+            get { return faultManager; }
+        }
+
+        class CaptureIfMessageSendingFails : IManageMessageFailures
+        {
+            static ILog Log = LogManager.GetLogger(typeof(CaptureIfMessageSendingFails));
+            private IDocumentStore store;
+            private IBus bus;
+            readonly Action executeOnFailure;
+
+            public CaptureIfMessageSendingFails(IDocumentStore store, IBus bus, Action executeOnFailure)
+            {
+                this.store = store;
+                this.bus = bus;
+                this.executeOnFailure = executeOnFailure;
+            }
+
+            public void SerializationFailedForMessage(TransportMessage message, Exception e)
+            {
+            }
+
+            public void ProcessingAlwaysFailsForMessage(TransportMessage message, Exception e)
+            {
+                try
+                {
+                    var destination = message.Headers["ServiceControl.TargetEndpointAddress"];
+                    var messageUniqueId = message.Headers["ServiceControl.Retry.UniqueMessageId"];
+                    Log.Error(string.Format("Failed to send '{0}' message to '{1}' for retry.", messageUniqueId, destination), e);
+
+                    var key = FailedMessage.MakeDocumentId(messageUniqueId);
+                    store.DatabaseCommands.Patch(key,
+                        new[]
+                        {
+                            new PatchRequest
+                            {
+                                Type = PatchCommandType.Set,
+                                Name = "Status",
+                                Value = (int) FailedMessageStatus.Unresolved,
+                            }
+                        });
+
+                    store.DatabaseCommands.Delete(FailedMessageRetry.MakeDocumentId(messageUniqueId), null);
+
+                    bus.Publish<MessagesSubmittedForRetryFailed>(m =>
+                    {
+                        m.FailedMessageId = messageUniqueId;
+                        m.Destination = destination;
+                        try
+                        {
+                            m.Reason = e.GetBaseException().Message;
+                        }
+                        catch (Exception)
+                        {
+                            m.Reason = "Failed to retrieve reason!";
+                        }
+
+                    });
+                }
+                catch (Exception ex)
+                {
+                    // If something goes wrong here we just ignore, not the end of the world!
+                    Log.Error("A failure occurred when trying to handle a retry failure.", ex);
```

Comment:
It doesn't block future retries from happening but it may leave an individual error message in an invalid state.